lib/Ohmbrewer_RIMS.cpp

```diff
@@ -28,57 +28,57 @@ Ohmbrewer::Pump* Ohmbrewer::RIMS::getRecirculator() const {
 /**
  * Constructor
  * @param id The Sprout ID to use for this piece of Equipment
- * @param pins The list of physical pins this Equipment is attached to
+ * @param tubePins[ temp busPin ; heating powerPin ; heating controlPin ]
+ * @param tunBus - mash tun temperature pin
+ * @param pumpPins[powerPin ; controlPin ]
  */
-Ohmbrewer::RIMS::RIMS(int id, std::list<int>* pins) : Ohmbrewer::Equipment(id, pins) {
-    // TODO: Figure out how to properly set the components of the RIMS the constructors
-    std::list<int>* fakePins = new std::list<int>(1,-1);
-    _tube = new Thermostat(1, fakePins);
-    _tunSensor = new TemperatureSensor(1, fakePins);
-    _recirc = new Pump(1,fakePins);
+Ohmbrewer::RIMS::RIMS(int id, int tubePins[], int tunBus, int pumpPins[] ) : Ohmbrewer::Equipment(id) {
+    _tube = new Thermostat(1, tubePins);
+    _tunSensor = new TemperatureSensor(1, tunBus);
```

Comment:
Assuming the TemperatureSensor is created after the Thermostat as is is here, and that there are no other TemperatureSensors in existence on this Rhizome, the ID would be '2' and not '1'. You'll want to fix this in other constructors too.